### PR TITLE
[FEAT] Add join_room and new_user sockets, to backend and frontend

### PR DESCRIPTION
**Pull request by Tianrens**
Basic lobby user joining. Requires that a game object has been created and game object has a session. If the session has a null host, the current user will become the host.

Added function to emit a join_room event to backend from the frontend.
Added function to listen ot join_room requests on backend, and allows the user to join the room and allocates the host.
Added sockets listening on new_user events emiting from backend to frontend.

### CODE REVIEW

File: frontend/src/sockets/socketClient.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {io} from 'socket.io-client';
```
Comment (Tianrens) [inline frontend/src/sockets/socketClient.js:1]: Should I change this to use export default?

File: backend/src/sockets/disconnect.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (afei088) [inline backend/src/sockets/disconnect.js:1]: not sure if we need a disconnect listener, its probably better to emit an event, and disconnect from the client side (e.g. from endGame method in the Game class)

File: frontend/src/sockets/socketClient.js (hunk 1 above)
Comment (afei088) [reply frontend/src/sockets/socketClient.js:1]: i personally think this is fine on the front end

File: backend/src/sockets/disconnect.js (hunk 2 above)
Comment (Tianrens) [reply backend/src/sockets/disconnect.js:1]: I think we need one. If a user closes the tab or "disconnects" some how we need to pick up on this event and delete the user from the Game.Session object. Or else when the user connects again, we'll get null errors.

Comment (afei088) [reply backend/src/sockets/disconnect.js:1]: true

**Review by afei088 — APPROVED** [review]

File: frontend/src/sockets/newUser.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Functino for when a new user joins a room.
```
Comment (afei088) [inline frontend/src/sockets/newUser.js:2]: Typo for Functiono

**Review by David-Xia0 — APPROVED** [review]

File: frontend/src/sockets/newUser.js (hunk 3 above)
Comment (Tianrens) [reply frontend/src/sockets/newUser.js]: fixed

**Review by David-Xia0 — DISMISSED** [review]

**Review by afei088 — DISMISSED** [review]

**Review by David-Xia0 — APPROVED** [review]

**Review by afei088 — APPROVED** [review]

**Pull request merged**